app/src/main/java/com/example/babytracking/SleepActivity.kt

```diff
@@ -1,11 +1,95 @@
 package com.example.babytracking
 
 import android.os.Bundle
+import android.widget.TimePicker
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.sleep_layout.sleep_button_save
+import kotlinx.android.synthetic.main.sleep_layout.sleep_check_lullaby
+import kotlinx.android.synthetic.main.sleep_layout.sleep_length
+import kotlinx.android.synthetic.main.sleep_layout.sleep_quality
+
+class SleepActivity : AppCompatActivity() {
+
+    private var hour: Int = 0
+    private var minute: Int = 0
+    private var ampm: String = "AM"
 
-class SleepActivity: AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.sleep_layout)
+
+        sleepTimeClick()
+        sleepButtonSave()
+    }
+
+    private fun sleepTimeClick() {
+        val timePicker = findViewById<TimePicker>(R.id.sleep_timepick_hour)
+        timePicker.setOnTimeChangedListener { _, pickerHour, pickerMinute ->
+            hour = pickerHour
+            minute = pickerMinute
+
+            // AM_PM decider logic
+            when {
```

Comment:
See comments in other file. Also, if you have the exact same logic in 2 places, could you consider sharing the logic somehow?

You could have a helper class to parse the hour / minute from the picker and return a timestamp, for example. So this class can be shared by both activities